app/services/link_directory.rb

```diff
@@ -0,0 +1,10 @@
+class LinkDirectory
+  def initialize(prison_finder:)
+    @prison_finder_base = prison_finder
+  end
+
+  def prison_finder(prison = nil)
+    slug = prison ? prison.finder_slug : nil
+    [@prison_finder_base, slug].compact.join('/')
```

Comment:
(Although I realise that will return a `URI`, not a string, so perhaps we need to do a `to_s` or similar.)